Add `entityMemberName` option to Metafix

fsteeg opened the pull request:
e.g. to support outputting array indexes as wildcards when listing available paths. Implemented like `arrayName` in `JsonDecoder`. Added tests in a new class due to NCSS checkstyle warning when expanding `MetafixRecordTest`.

Will resolve #294.

blackwinter reviewed (approved):
> Added tests in a new class due to NCSS checkstyle warning when expanding `MetafixRecordTest`.

We've already disabled this check in some cases, maybe it's even unnecessary for test classes.

In general, I'd prefer each class file's tests to live in a correspondingly named test file (`SomeClass.java` -> `SomeClassTest.java`). It's currently not the case and that's sometimes a hindrance to finding a particular test or, even more so, to finding the appropriate place for new tests.

In this particular case: The original tests for `setRepeatedFieldsToEntities()` are still in `MetafixRecordTest.java`, but the new tests for `setEntityMemberName()` (and, coincidentally, for `setRepeatedFieldsToEntities()`) are in `MetafixConfigTest.java`.

I won't object, but I'm feeling a little uneasy about these scattered tests...

fsteeg commented:
Thanks for the quick review! I've addressed your feedback in https://github.com/metafacture/metafacture-fix/pull/299/commits/935f05fea639137514df3ffe0c7f29b78591681e.

blackwinter commented:
Great, thanks :)
